File: WPILib/src/main/java/com/github/ezauton/wpilib/sensors/Gyros.java
package com.github.ezauton.wpilib.sensors;

import com.github.ezauton.core.localization.sensors.ICompass;
import com.kauailabs.navx.frc.AHRS;

public class Gyros
{
    public static ICompass fromNavx(AHRS navx)
    {
        return () -> {
            double angle = -navx.getAngle(); // we want CCW orientation
            double boundedAngle = angle % 360;
            if(boundedAngle < 0)
            {
                boundedAngle = 360 + boundedAngle;
            }
            return boundedAngle;
        };
    }
}

File: WPILib/src/main/kotlin/com/github/ezauton/wpilib/sensors/Gyros.kt
package com.github.ezauton.wpilib.sensors

import com.kauailabs.navx.frc.AHRS
import com.github.ezauton.core.localization.sensors.Compass

object Gyros {
  fun fromNavx(navx: AHRS): Compass {
    return Compass {
      val angle = -navx.angle // we want CCW orientation
      var boundedAngle = angle % 360
      if (boundedAngle < 0) {
        boundedAngle = 360 + boundedAngle
      }
      boundedAngle
    }
  }
}
